metagraph/tests/graph/all/test_dbg_helpers.cpp

```diff
@@ -389,6 +406,21 @@ build_graph_batch<DBGSuccinctBloom<4, 50>>(uint64_t k,
     return graph;
 }
 
+template <>
+std::shared_ptr<DeBruijnGraph>
+build_graph_batch<DBGSuccinct::CachedView>(uint64_t k,
+                                           std::vector<std::string> sequences,
+                                           DeBruijnGraph::Mode mode) {
+    auto base_graph = std::dynamic_pointer_cast<DBGSuccinct>(
+        build_graph_batch<DBGSuccinct>(k, sequences, mode)
+    );
+    auto cached = base_graph->get_cached_view();
+
+    // reset the underlying pointer to use base_graph instead of an alias of it
+    cached->set_graph(base_graph);
+    return cached;
+}
+
```

Comment:
The cached_view doesn't care how the underlying graph was constructed.

Thus, only one of `build_graph<DBGSuccinct::CachedView>` and `build_graph_batch<DBGSuccinct::CachedView>` should exist.

Reply:
Having separate methods is useful because it can test that the CachedView works for graphs with (`build_graph`) and without (`build_graph_batch`) redundant dummy edges.